unit/src/ParsedFunctionTest.C

```diff
@@ -34,10 +34,12 @@ ParsedFunctionTest::init()
 
   InputParameters mesh_params = _factory->getValidParams("GeneratedMesh");
   mesh_params.set<MooseEnum>("dim") = "3";
+  mesh_params.set<std::string>("name") = "mesh";
```

Comment:
Ok the later is important because 99% of people won't go around the Factory, but people building objects in an Action shouldn't have to set the name.